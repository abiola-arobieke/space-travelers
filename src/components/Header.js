import { NavLink } from 'react-router-dom';

const Header = () => (
  <>
    <nav className="nav">
      <div className="name-logo-container">
        <img className="logo" src="https://i.postimg.cc/x8xhGBMh/planet.png" alt="logo" />
        <NavLink className="name-logo" to="/">Space Travelers&apos; Hub</NavLink>
      </div>
      <ul className="ul">
        <li className="li">
          <NavLink className="navlink" to="/">Rockets</NavLink>
        </li>
        <li className="li">
          <NavLink className="navlink missions-navlink" to="/missions">Missions</NavLink>
        </li>
        <li className="li">
          <NavLink className="navlink" to="/profile">My Profile</NavLink>
        </li>
      </ul>
    </nav>
  </>
);

export default Header;
