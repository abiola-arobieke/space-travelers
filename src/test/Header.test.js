import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Header from '../components/Header';

describe('Header component', () => {
  test('renders header with navigation links', () => {
    const { container } = render(
      <MemoryRouter>
        <Header />
      </MemoryRouter>,
    );

    expect(container).toMatchSnapshot();
  });
});
