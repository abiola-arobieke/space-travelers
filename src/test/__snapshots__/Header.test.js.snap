// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders header with navigation links 1`] = `
<div>
  <nav
    class="nav"
  >
    <div
      class="name-logo-container"
    >
      <a
        aria-current="page"
        class="active"
        href="/"
      >
        <img
          alt="logo"
          class="logo"
          src="https://i.postimg.cc/x8xhGBMh/planet.png"
        />
      </a>
      <a
        aria-current="page"
        class="name-logo active"
        href="/"
      >
        Space Travelers' Hub
      </a>
    </div>
    <ul
      class="ul"
    >
      <li
        class="li"
      >
        <a
          activeclassname="active"
          aria-current="page"
          class="navlink active"
          href="/"
        >
          Rockets
        </a>
      </li>
      <li
        class="li"
      >
        <a
          activeclassname="active"
          class="navlink missions-navlink"
          href="/missions"
        >
          Missions
        </a>
      </li>
      <li
        class="li"
      >
        <a
          activeclassname="active"
          class="navlink"
          href="/profile"
        >
          My Profile
        </a>
      </li>
    </ul>
  </nav>
</div>
`;
